README
NAME
    Dist::Zilla::Plugin::Prereqs::MatchInstalled::All - Upgrade ALL your
    dependencies to the ones you have installed.

VERSION
    version 1.001001

SYNOPSIS
        [Prereqs::MatchInstalled::All]
        ; upgrade_perl = 1  ; if you want to upgrade to your installed perl
        ; include these too if you don't want to force a perl upgrade indirectly.
        exclude = strict
        exclude = warnings

DESCRIPTION
    This is a special case of "Dist::Zilla::Plugin::Prereqs::MatchInstalled"
    that automatically upgrades all versions of all dependencies, unless
    asked not to.

  PITFALLS
    Presently, there is one very large gotcha about using this module, in
    that it will upgrade everything, even things that don't make sense to
    upgrade.

    For instance:

   Local Versions
    If you have a single dependency on your system you might use, which is
    locally patched, and locally patched in such a way the local version is
    more recent than any on "CPAN", you should either

    a. Not use this module
    b. Put that module in the exclusion list

   Non-Dual Life modules
    This plugin is not very smart, and can't differentiate between modules
    that do exist on "CPAN" independent of Perl, and modules that don't.

    For instance, if you use "Autoprereqs", its very likely your
    distribution will add a dependency on either "strict" or "warnings"

    This module will ask your user to upgrade those versions to their latest
    versions, which will likely require them to upgrade their Perl
    installation to do so.

    Which basically means for the mean time, either

    a. You must be o.k. with end users needing more recent Perls
    b. You should avoid upgrading those dependencies by either

        a. Not using this plugin
        b. Adding problematic modules to the exclusion list

ATTRIBUTES
  "exclude"
    This parameter can be specified multiple times, and each time should
    represent a single package string to exclude from version upgrades.

        [Prereqs::MatchInstalled::All]
        exclude = foo
        exclude = bar

  "upgrade_perl"
    If specified, this will permit upgrades on the dependency on "perl" to
    the installed version.

        [Prereqs::MatchInstalled::All]
        upgrade_perl = 1

    Note, this has no effect on the modules that may inherently be only
    available by upgrading Perl.

    Default is false.

    See "PITFALLS" for details.

PRIVATE ATTRIBUTES
  "_exclude_hash"
PRIVATE METHODS
  "_build__exclude_hash"
  "_user_wants_excluded"
  "_user_wants_upgrade_on"
AUTHOR
    Kent Fredric <kentnl@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Kent Fredric
    <kentfredric@gmail.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Release history for Dist-Zilla-Plugin-Prereqs-MatchInstalled-All

1.001002 2017-03-08T10:42:36Z cce52f5

1.001001 2017-03-08T10:40:55Z de83365
 - Bugfix: Avoid test failures with -Ddefault_inc_excludes_dot
 - Drop use of ConfigDumper Util
 - Drop use of Test::KENTNL/dztest

 [Dependencies::Stats]
 - Dependencies changed since 1.001000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1, suggests: +1)
 - develop: +12 ↑3 -4 (recommends: +1, suggests: ↑2)
 - runtime: -1 (recommends: +1)
 - test: +1 ↓1 -1 (recommends: ↑1 ↓1)

1.001000 2014-10-15T22:42:03Z f59f09b
 [Dependencies::Stats]
 - Dependencies changed since 1.000000, see misc/*.deps* for details
 - configure: ↓1
 - develop: +8 ↑2 -3 (suggests: +2 -1)
 - runtime: +1
 - test: +2 -7 (recommends: ↑2 -1)

 [Internals]
 - Implement dump_config with config_dumper
 - Implement tests with dztest

 [Metadata]
 - Show 1x Contributor
 - Move to Kentnl org
 - Set author=kentnl@cpan.org

1.000000 2014-05-02T09:30:17Z
 [00 Minor]
 - No important code changes
 - Just packaging changes downstream vendors who work by hand may have to worry about.

 [01 Toolkit Change]
 - Uses ExtUtils::MakeMaker instead of Module::Build
 - Dependencies softened ( aka: No longer building with ourselves )

 [02 Version scheme change]
 - now uses x.123456 instead of x.y.z
 - this is the primary reason for the 1.x on the box.

 [Dependencies::Aggregate]
 - 5.008009 : ? → 205
 - 5.010001 : ? → 184
 - 5.018000 : 163 → ?
 - 5.018002 : ? → 170
 - 5.019000 : 163 → ?
 - 5.019009 : ? → 165

 [Dependencies::Stats]
 - Dependencies changed since 0.1.2, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +42 ↓5 -2 (recommends: -1, suggests: +1 -1)
 - runtime: +1 ↑1 ↓3
 - test: +5 ↓5 -5 (recommends: +4)

0.1.2 2013-09-25T13:43:34Z
 [00 Minor/Bugfix]
 - Rebuild to get latest Test::Compile test which is not broken on Win32
 - Added one basic test

 [Dependencies::Stats]
 - Dependencies changed since 0.1.1, see misc/*.deps* for details
 - develop: ↑3 (recommends: ↑1)
 - runtime: ↑2
 - test: +7

0.1.1 2013-08-31T02:39:00Z
 [!Minor]
 - Rebuild Tests w/ newer Test::Compile

 [Dependencies::Stats]
 - Dependencies changed since 0.1.0, see misc/*.deps* for details
 - develop: +1 ↑2 (suggests: ↑1)
 - test: +2 -2

0.1.0 2013-08-07T23:27:23Z
 - First version.


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4007"
  },
  "configure_requires" => {
    "Module::Build" => "0.4007"
  },
  "dist_abstract" => "Upgrade ALL your dependencies to the ones you have installed.",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-Prereqs-MatchInstalled-All",
  "dist_version" => "0.1.3",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::Prereqs::MatchInstalled::All",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla::Plugin::Prereqs::MatchInstalled" => "v0.1.4",
    "Moose" => "2.1005",
    "MooseX::Types::Moose" => "0.38",
    "perl" => "v5.10.0",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Cwd" => 0,
    "File::Copy::Recursive" => "0.38",
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Path::FindDev" => "v0.3.1",
    "Path::Tiny" => "0.037",
    "Test::DZil" => "4.300039",
    "Test::Fatal" => "0.013",
    "Test::More" => "0.98"
  }
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  my $tr = delete $module_build_args{test_requires};
  my $br = $module_build_args{build_requires};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Upgrade ALL your dependencies to the ones you have installed.",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Prereqs-MatchInstalled-All",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008",
  "NAME" => "Dist::Zilla::Plugin::Prereqs::MatchInstalled::All",
  "PREREQ_PM" => {
    "Dist::Zilla::Plugin::Prereqs::MatchInstalled" => "1.000000",
    "Moose" => 0,
    "MooseX::Types::Moose" => 0,
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::DZil" => 0,
    "Test::More" => 0
  },
  "VERSION" => "1.001002",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::Plugin::Prereqs::MatchInstalled" => "1.000000",
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Moose" => 0,
  "MooseX::Types::Moose" => 0,
  "Test::DZil" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
